server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/DeadCompactionDetector.java

```diff
@@ -84,21 +97,42 @@ private void detectDeadCompactions() {
 
     running.forEach((ecid) -> {
       if (tabletCompactions.remove(ecid) != null) {
-        log.trace("Removed {} running on a compactor", ecid);
+        log.trace("Removed compaction {} running on a compactor", ecid);
+      }
+      if (this.deadCompactions.remove(ecid) != null) {
+        log.trace("Removed {} from the dead compaction map, it's running on a compactor", ecid);
       }
     });
 
     // Determine which compactions are currently committing and remove those
     context.getAmple().getExternalCompactionFinalStates()
-        .map(ecfs -> ecfs.getExternalCompactionId()).forEach(tabletCompactions::remove);
+        .map(ecfs -> ecfs.getExternalCompactionId()).forEach(ecid -> {
+          if (tabletCompactions.remove(ecid) != null) {
+            log.trace("Removed compaction {} that is committing", ecid);
+          }
+          if (this.deadCompactions.remove(ecid) != null) {
+            log.trace("Removed {} from the dead compaction map, it's committing", ecid);
+          }
+        });
 
-    tabletCompactions
-        .forEach((ecid, extent) -> log.debug("Detected dead compaction {} {}", ecid, extent));
+    tabletCompactions.forEach((ecid, extent) -> {
+      log.debug("Possible dead compaction detected {} {}", ecid, extent);
+      this.deadCompactions.putIfAbsent(ecid, System.currentTimeMillis());
+      this.deadCompactions.merge(ecid, 1L, Long::sum);
+    });
 
     // Everything left in tabletCompactions is no longer running anywhere and should be failed.
     // Its possible that a compaction committed while going through the steps above, if so then
     // that is ok and marking it failed will end up being a no-op.
+    Set<ExternalCompactionId> toFail =
+        this.deadCompactions.entrySet().stream().filter(e -> e.getValue() > 2).map(e -> e.getKey())
+            .collect(Collectors.toCollection(TreeSet::new));
+    tabletCompactions.keySet().retainAll(toFail);
+    tabletCompactions.forEach((eci, v) -> {
+      log.warn("Compaction {} believed to be dead, failing it.", eci);
```

Comment:
Maybe make this info.  Thinking that compactor processes dying will be somewhat normal/expected so may not want to warn.

```suggestion
      log.info("Compaction {} believed to be dead, failing it.", eci);
```

Reply:
I think compactor processes dying might only be normal in the situation where you are dynamically scaling and the scaling mechanism has no way to gracefully stop the compactor process. In a static deployment, compactor processes should not be dying. I think `warn` might be appropriate as it's highlighting a failure. Curious what others have to say here.

Reply:
Yeah I can see that case for making it warn.  I feel it should be INFO or WARN, but not sure which. WARN is fine w/ me.

Reply:
Why would that be expected?  If something is repeatedly dying I'd think that would indicate that some action / investigation might be warranted.  If I had a compaction that was taking a long time - and it was because these kept dying, how would I know?

Recently had a discussion  related to the trace level statements that are associated with this particular log message, wondering if they might not be better at debug (at least for some of them) - my impression is that dead / dying and removing the entries should not be common - and if they are then something should be fixed so they are not.   It might not be obvious that if you are having issues with external compaction you need to set logging to trace to obtain enough info to figure out what the issue might be.  I realize that we often log to much, but from looking at the code without understanding the frequency / how common / what a corrective approach might be - some of this code might benefit from more logging at debug or higher levels - just my initial impression, would probability need to run and then have to actually try to debug something before having something concrete to add to the discussion.

Reply:
> Why would that be expected?

I was thinking that on a large cluster, host and processes dying is expected.  In general I think Accumulo has a bit too much logging at the warn level for things that occur as a routine part of operating a cluster.  If so, maybe this makes people ignore warnings.

>  It might not be obvious that if you are having issues with external compaction you need to set logging to trace to obtain enough info to figure out what the issue might be.

I was thinking that detection of a dead compaction should be at WARN or INFO, but was not sure which after @dlmarion comment.  The stuff at trace is related to the source information on how it finds dead compactions and will routinely log information even when it does not detect a dead compaction.

Reply:
I like how the change logs at debug when it finds a candidate dead compaction and adds that to the map.